keanu-python/keanu/context.py

```diff
@@ -66,15 +67,19 @@ def to_java_map(self, python_map: Dict[Any, Any]) -> JavaMap:
         m = self._gateway.jvm.java.util.HashMap()
 
         for (k, v) in python_map.items():
-            m.put(k.unwrap(), v.unwrap())
+            new_k = k.unwrap() if isinstance(k, JavaObjectWrapper) else k
+            new_v = v.unwrap() if isinstance(v, JavaObjectWrapper) else v
```

Comment:
Other ways we could do this - not as neat but avoids the dependency on `base`:
* before you call this method, wrap your `str` objects in a class that has an `unwrap` method
* check for the opposite case e.g. `isinstance(v, runtime_primitive_types) or isinstance(v, str)`
* try/except

Reply:
I'll have a think, but as you say - definitely not as clean and/or a lot more brittle.